Keen.NetStandard/KeenClient.cs

```diff
@@ -972,5 +973,177 @@ public JObject Query(string queryName, Dictionary<string, string> parms)
                 throw ex.TryUnwrap();
             }
         }
+
+        /// <summary>
+        /// Get query results from a Cached Dataset.
+        /// </summary>
+        /// <param name="datasetName">Name of cached dataset to query.</param>
+        /// <param name="indexBy">The string property value you want to retrieve results by.</param>
+        /// <param name="timeframe">Limits retrieval of results to a specific portion of the Cached Dataset</param>
+        /// <returns>An instance of Newtonsoft.Json.Linq.JObject containing query results and metadata defining the cached dataset.</returns>
+        public async Task<JObject> QueryDatasetAsync(string datasetName, string indexBy, string timeframe)
+        {
+            return await Datasets.GetResultsAsync(datasetName, indexBy, timeframe);
+        }
+
+        /// <summary>
+        /// Get query results from a Cached Dataset.
+        /// </summary>
+        /// <param name="datasetName">Name of cached dataset to query.</param>
+        /// <param name="indexBy">The string property value you want to retrieve results by.</param>
+        /// <param name="timeframe">Limits retrieval of results to a specific portion of the Cached Dataset</param>
+        /// <returns>An instance of Newtonsoft.Json.Linq.JObject containing query results and metadata defining the cached dataset.</returns>
+        public JObject QueryDataset(string datasetName, string indexBy, string timeframe)
+        {
+            try
+            {
+                return QueryDatasetAsync(datasetName, indexBy, timeframe).Result;
+            }
+            catch (AggregateException ex)
+            {
+                throw ex.TryUnwrap();
+            }
+        }
+
+        /// <summary>
+        /// Get the definition of your cached dataset.
+        /// </summary>
+        /// <param name="datasetName">Name of cached dataset to get the definition of.</param>
+        /// <returns>An instance of Keen.Core.Dataset.DatasetDefinition containing metadata about your cached dataset.</returns>
+        public async Task<DatasetDefinition> GetDatasetDefinitionAsync(string datasetName)
+        {
+            return await Datasets.GetDefinitionAsync(datasetName);
```

Comment:
This needs to call `.ConfigureAwait(false)`

Reply:
Here and elsewhere I just return the `Task` directly since there's no need to `await` and no need to configure an awaiter--in any case internally our impls configure the awaiter, and if 3rd party impls don't, there's not much we can do about it.